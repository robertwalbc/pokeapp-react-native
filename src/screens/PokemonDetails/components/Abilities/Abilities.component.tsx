import React from 'react';
import { View, Text } from 'react-native';
import { styles } from './Abilities.styles.ts';

interface Ability {
  ability: {
    name: string;
  };
}

interface AbilitiesProps {
  abilities: Ability[];
}

export const Abilities: React.FC<AbilitiesProps> = ({ abilities }) => (
  <>
    <Text style={styles.sectionTitle}>Abilities</Text>
    <View style={styles.abilitiesContainer}>
      {abilities.map((abilityInfo, index) => (
        <View key={index} style={styles.abilityChip}>
          <Text style={styles.abilityText}>
            {abilityInfo.ability.name.replace('-', ' ')}
          </Text>
        </View>
      ))}
    </View>
  </>
); 