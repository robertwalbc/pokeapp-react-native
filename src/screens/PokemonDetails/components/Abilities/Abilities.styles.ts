import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  sectionTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    marginTop: 16,
    marginBottom: 8,
  },
  abilitiesContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    marginBottom: 16,
  },
  abilityChip: {
    backgroundColor: '#E0E0E0',
    paddingHorizontal: 16,
    paddingVertical: 6,
    borderRadius: 16,
    margin: 4,
  },
  abilityText: {
    fontSize: 14,
    textTransform: 'capitalize',
  },
}); 