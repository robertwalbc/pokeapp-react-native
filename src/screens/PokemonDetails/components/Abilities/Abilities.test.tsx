import React from 'react';
import renderer from 'react-test-renderer';
import { Abilities } from './Abilities.component';

describe('Abilities', () => {
  it('renders the section title and ability names', () => {
    const abilities = [
      { ability: { name: 'overgrow' } },
      { ability: { name: 'chlorophyll' } },
    ];
    const tree = renderer.create(<Abilities abilities={abilities} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 