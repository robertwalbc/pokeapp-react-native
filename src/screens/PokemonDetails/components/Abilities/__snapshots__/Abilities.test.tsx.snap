// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abilities renders the section title and ability names 1`] = `null`;
